yt_dlp/utils/traversal.py

```diff
@@ -247,6 +265,10 @@ def apply_path(start_obj, path, test_type):
                     objs = (list(filtered_objs),)
                 continue
 
+            if key is filter:
+                obj = filter(None, objs)
```

Comment:
```suggestion
                objs = filter(None, objs)
```
At least that works upstream and seems to match the other cases.

Reply:
Ill have to update tests to catch this

Reply:
ISTR my ported test_traversal.py failing on this.